Assets/_Laz/Scripts/Environment/Puzzles/LazoActivation.cs

```diff
@@ -0,0 +1,58 @@
+using System.Linq;
+
+namespace Laz
+{
+    [System.Serializable]
+    public struct LazoActivation
+    {
+        public BasePuzzleBehaviour[] PuzzleItems;
+        public BaseCompletedPuzzleActivationBehaviour ItemToActivate;
+
+        public void Initialize()
+        {
+            foreach (var puzzleItem in PuzzleItems)
+            {
+                puzzleItem.Initialize();
+                puzzleItem.OnPuzzleCompleted += ActivateWrapItemIfNeeded;
+            }
+            
+            ItemToActivate.Initialize();
+        }
+
+        public void CleanUp()
+        {
+            foreach (var item in PuzzleItems)
+            {
+                item.CleanUp();
+            }
+            ItemToActivate.CleanUp();
+        }
+
+        public void Reset()
+        {
+            foreach (var item in PuzzleItems)
+            {
+                item.Reset();
+            }
+            ItemToActivate.Reset();
+        }
+
+        private void ActivateWrapItemIfNeeded()
+        {
+            if (AreAllPuzzlesActivated())
+            {
+                ActivatePuzzleItem();
+            }
+        }
+
+        private bool AreAllPuzzlesActivated()
+        {
+            return PuzzleItems.All(item => item.IsActivated);
+        }
```

Comment:
not sure how often ActivateWrapItemIfNeeded is called but if it is frequency might be best to not use Linq?

Reply:
It's not called every frame, just when a set of puzzles are completed. E.g. Planetoid get's wrapped or
Series of Nodes get's activated